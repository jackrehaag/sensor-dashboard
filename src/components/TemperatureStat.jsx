import React from "react";

const TemperatureStat = ({ temp }) => {
  let color;
  if (temp < 10) {
    color = "blue";
  } else if (temp >= 10 && temp < 20) {
    color = "yellow";
  } else if (temp >= 20 && temp < 30) {
    color = "orange";
  } else if (temp >= 30) {
    color = "red";
  } else {
    color = "black";
  }
  return (
    <div className={color + " current-temp statistic"}>
      <div className="value">{temp}Â°C</div>
      <div className="label">
        Temperature <i className="thermometer half icon" />
      </div>
    </div>
  );
};

export default TemperatureStat;
