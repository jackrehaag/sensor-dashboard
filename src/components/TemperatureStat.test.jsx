import React from "react";
import TemperatureStat from "./TemperatureStat";
import renderer from "react-test-renderer";

describe("changing the color class based on temperature prop", () => {
  test("it sets class to blue when temperature is below 10", () => {
    const component = renderer.create(<TemperatureStat temp={1} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to blue when temperature is below 10 (minus value)", () => {
    const component = renderer.create(<TemperatureStat temp={-3} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to yellow when temperature is 10", () => {
    const component = renderer.create(<TemperatureStat temp={10} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to yellow when temperature is above 10", () => {
    const component = renderer.create(<TemperatureStat temp={11} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to orange when temperature is 20", () => {
    const component = renderer.create(<TemperatureStat temp={20} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to orange when temperature is above 20", () => {
    const component = renderer.create(<TemperatureStat temp={22} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to red when temperature is 30", () => {
    const component = renderer.create(<TemperatureStat temp={30} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to red when temperature is above 30", () => {
    const component = renderer.create(<TemperatureStat temp={35} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("it sets class to black when temperature is NaN", () => {
    const component = renderer.create(<TemperatureStat temp={"NaN"} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
