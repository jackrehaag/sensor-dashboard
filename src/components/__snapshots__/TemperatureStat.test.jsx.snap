// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changing the color class based on temperature prop it sets class to black when temperature is NaN 1`] = `
<div
  className="black current-temp statistic"
>
  <div
    className="value"
  >
    NaN
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to blue when temperature is below 10 (minus value) 1`] = `
<div
  className="blue current-temp statistic"
>
  <div
    className="value"
  >
    -3
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to blue when temperature is below 10 1`] = `
<div
  className="blue current-temp statistic"
>
  <div
    className="value"
  >
    1
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to orange when temperature is 20 1`] = `
<div
  className="orange current-temp statistic"
>
  <div
    className="value"
  >
    20
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to orange when temperature is above 20 1`] = `
<div
  className="orange current-temp statistic"
>
  <div
    className="value"
  >
    22
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to red when temperature is 30 1`] = `
<div
  className="red current-temp statistic"
>
  <div
    className="value"
  >
    30
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to red when temperature is above 30 1`] = `
<div
  className="red current-temp statistic"
>
  <div
    className="value"
  >
    35
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to yellow when temperature is 10 1`] = `
<div
  className="yellow current-temp statistic"
>
  <div
    className="value"
  >
    10
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;

exports[`changing the color class based on temperature prop it sets class to yellow when temperature is above 10 1`] = `
<div
  className="yellow current-temp statistic"
>
  <div
    className="value"
  >
    11
    °C
  </div>
  <div
    className="label"
  >
    Temperature 
    <i
      className="thermometer half icon"
    />
  </div>
</div>
`;
